src/test/java/com/iota/iri/integration/APIIntegrationTests.java

```diff
@@ -396,27 +400,60 @@ public void shouldSendTransactionAndFetch() throws InterruptedException {
         request.clear();
         request.put("command", "storeTransactions");
         request.put("trytes", trytes);
-
         given().
                 body(gson().toJson(request)).
                 when().
                 post("/").
                 then().
                 log().all().and().spec(responseSpec);
 
-        //Fetch
-        String temp = (String) trytes.get(0);
-        String[] addresses = {temp.substring(2187,2268)}; //extract address from trytes
+        return trytes;
+    }
+
+    private List<Object> findTransactions(String key, String[] values) {
+        final Map<String, Object> request = new HashMap<>();
         request.clear();
         request.put("command", "findTransactions");
-        request.put("addresses", addresses);
-        response = given().
+        request.put(key, values);
+        Response response = given().
                 body(gson().toJson(request)).
                 when().
                 post("/");
         response.getBody();
-        responseJson = response.jsonPath();
-        List<Object> hashes = responseJson.getList("hashes");
+        JsonPath responseJson = response.jsonPath();
+
+        return responseJson.getList("hashes");
+    }
+
+    @Test
+    public void shouldSendTransactionAndFetchByAddress() {
+
+        List<Object> trytes = sendTransfer(TRYTES);
+
+        String temp = (String) trytes.get(0);
+        String[] addresses = {temp.substring(TransactionViewModel.ADDRESS_TRINARY_OFFSET / 3,
+                (TransactionViewModel.ADDRESS_TRINARY_OFFSET + TransactionViewModel.ADDRESS_TRINARY_SIZE) / 3)}; //extract address from trytes
+        List<Object> hashes = findTransactions("addresses", addresses);
         Assert.assertTrue(!hashes.isEmpty());
     }
+
+    @Test
+    public void shouldSendTransactionAndFetchByTag() {
+
+        List<Object> trytes = sendTransfer(TRYTES);
+
+        //Tag
+        String temp = (String) trytes.get(0);
+        String[] tags = {temp.substring(TransactionViewModel.TAG_TRINARY_OFFSET / 3,
+                (TransactionViewModel.TAG_TRINARY_OFFSET + TransactionViewModel.TAG_TRINARY_SIZE) / 3)}; //extract address from trytes
+        List<Object> hashes = findTransactions("tags", tags);
+        Assert.assertTrue(!hashes.isEmpty());
```

Comment:
is it possible to use `assertEquals` since you know what the tag will be?

Reply:
well, the result of "findTransactions" is the hash of the transaction.
which we don't know a priori, the `trytes` can be hashed to find the transaction's hash and then compared -- adding.